### PR TITLE
General Cleanup 7

### PR DESCRIPTION
**Pull request by hensldm**
Some more cleanup around the code base. As part of this, fixes the two incorrect render mode defines mentioned in https://github.com/zeldaret/oot/pull/1313.

### CODE REVIEW

File: src/overlays/actors/ovl_Dm_Char00/z_dm_char00.c [hunk 1]
```diff
@@ -934,13 +934,13 @@ void DmChar00_Draw(Actor* thisx, PlayState* play2) {
...
-                    (u8)(this->unk_240.unk_0C * 1));
+    gDPSetPrimColor(gfx++, 0, 0x01, (u8)this->unk_240.r, (u8)this->unk_240.g, (u8)this->unk_240.b,
+                    (u8)(this->unk_240.a * 1));
     gDPSetRenderMode(gfx++, G_RM_PASS, G_RM_ZB_CLD_SURF2);
     gSPEndDisplayList(gfx);
 
-    gDPSetEnvColor(POLY_XLU_DISP++, (u8)(s8)this->unk_250.unk_00, (u8)(s8)this->unk_250.unk_04,
-                   (u8)(s8)this->unk_250.unk_08, (u8)(s8)((f32)phi_a0 * 1));
+    gDPSetEnvColor(POLY_XLU_DISP++, (u8)(s8)this->unk_250.r, (u8)(s8)this->unk_250.g, (u8)(s8)this->unk_250.b,
+                   (u8)(s8)((f32)phi_a0 * 1));
```
Comment (mzxrules) [inline src/overlays/actors/ovl_Dm_Char00/z_dm_char00.c:943]: how necessary are these casts?

File: src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.h [hunk 2]
```diff
@@ -33,9 +33,9 @@ typedef struct EnKakasi {
     /* 0x20C */ f32 unk20C; // passed to unknown function
     /* 0x210 */ f32 unk210; // used as a target for 20C to approach
     /* 0x214 */ Vec3f unk214; // copied from unk238 regularly
-    /* 0x220 */ Vec3f unk220; //EnKakasi_TeachingSong
-    /* 0x22C */ Vec3f unk22C; //actor home copied to here
-    /* 0x238 */ Vec3f unk238; //copied from D_80971E38[unk190]
+    /* 0x220 */ Vec3f unk220; // EnKakasi_TeachingSong
+    /* 0x22C */ Vec3f unk22C; // actor home copied to here
+    /* 0x238 */ Vec3f unk238; // copied from D_80971E38[unk190]
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.h:38]: I would just not bother with these, these are all subCam variables named in another open PR. I'll just go ahead and delete these comments here and save a conflict

File: src/overlays/gamestates/ovl_title/z_title.c [hunk 3]
```diff
@@ -91,7 +91,7 @@ void Title_Draw(GameState* thisx) {
 
     gDPPipeSync(POLY_OPA_DISP++);
     gDPSetCycleType(POLY_OPA_DISP++, G_CYC_2CYCLE);
-    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF2, G_RM_OPA_CI | CVG_DST_WRAP);
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_PASS, G_RM_XLU_SURF2 | CVG_DST_WRAP);
```
Comment (Dragorn421) [inline src/overlays/gamestates/ovl_title/z_title.c:94]:
```suggestion
    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_PASS, G_RM_CLD_SURF2);
```
( https://github.com/zeldaret/oot/pull/1313#discussion_r918950375 )

File: src/overlays/actors/ovl_Dm_Char00/z_dm_char00.c (hunk 1 above)
Comment (hensldm) [reply src/overlays/actors/ovl_Dm_Char00/z_dm_char00.c:943]: They all seem required to match.

File: src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.h (hunk 2 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.h:38]: Ahh, I'll just remove this from the PR than.

File: src/overlays/actors/ovl_En_Shn/z_en_shn.h [hunk 4]
```diff
@@ -7,6 +7,7 @@
 struct EnShn;
 
 typedef void (*EnShnActionFunc)(struct EnShn*, PlayState*);
+typedef s32 (*MsgEventFunc)(Actor*, PlayState*);
```
[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/actors/ovl_En_Shn/z_en_shn.h:10]: Hmm, there seems to be quite a few of these. In order to not to grow this PR too much, what about I revert this change and just do them all in a separate PR?

**Review by engineer124 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**